ansible_rulebook/engine.py

```diff
@@ -281,12 +321,22 @@ async def run_rulesets(
             logger.info("Cancelling " + task.get_name())
             task.cancel()
 
+    should_reload = False
+    if (
+        monitor_task
+        and monitor_task.exception().__class__ == DevReloadException
```

Comment:
The right way I think it would be:
```suggestion
        and isinstance(monitor_task.exception(), DevReloadException)
```